app/controllers/schemes_controller.rb

```diff
@@ -44,8 +44,15 @@ def deactivate_confirm
   end
 
   def deactivate
-    if @scheme.scheme_deactivation_periods.create!(deactivation_date: params[:deactivation_date]) && reset_location_and_scheme_for_logs!
+    if @scheme.scheme_deactivation_periods.create!(deactivation_date: params[:deactivation_date])
+      logs = reset_location_and_scheme_for_logs!
```

Comment:
(See `location_controller` comment for why I'm okay with moving this out of the condition.)